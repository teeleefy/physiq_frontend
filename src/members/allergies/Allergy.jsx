import {
    Card,
    CardBody,
    CardText,
    ListGroupItem
  } from "reactstrap";
  import { NavLink } from "react-router-dom";
import '../styles/Allergy.css'


function Allergy({allergy, memberId}){

    return(
        <>
            <Card className="Allergy-card">
                <CardBody>
                <CardText>
                    <p><b>Allergy Name:</b> <NavLink className="Allergy-name" to={`/member/${memberId}/allergies/${allergy.id}`} key={allergy.id}>
                        {allergy.name}
                    </NavLink></p>
                    <p><b>Reaction:</b> {allergy.reaction}</p>
                    <p><b>Notes:</b> {allergy.notes}</p>
                </CardText>
                </CardBody>
            </Card>
        </>
     )
}

export default Allergy;