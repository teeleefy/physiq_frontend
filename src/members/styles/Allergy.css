.Allergy-card {
    height: fit-content;
    margin-left: 0.5rem;
    margin-bottom: 0.5rem;
}

.Allergy-label {
    font-size: 1.5rem;
}

.Allergy-h1 {
    font-size: 3rem;
    font-weight: 600;
}

.Allergies-navlink {
    text-decoration: none;
    color: white;
}