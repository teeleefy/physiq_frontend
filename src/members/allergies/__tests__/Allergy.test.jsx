import { renderWithRouter, screen } from '../../../../__tests__/test-tools';
import Allergy from '../Allergy'

let allergy={name: "bees", reaction: "hives", notes: "Carry epipen."}
let memberId=1000;

describe('Allergy', () => {
  it('renders the Allergy component', () => {
    renderWithRouter(<Allergy allergy={allergy} memberId={memberId}/>)
    
    screen.debug(); // prints out the jsx in the Allergy component unto the command line
  })
})

it("matches snapshot", function() {
  const { asFragment } = renderWithRouter(<Allergy allergy={allergy} memberId={memberId}/>);
  expect(asFragment()).toMatchSnapshot();
});