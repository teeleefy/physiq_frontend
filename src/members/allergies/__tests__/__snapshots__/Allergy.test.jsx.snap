// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="list-group-item"
  >
    <div
      class="Allergy-card card"
    >
      <div
        class="card-body"
      >
        <p
          class="card-text"
        />
        <p>
          <b>
            Allergy Name:
          </b>
           
          <a
            class="Allergy-name"
            href="/member/1000/allergies/undefined"
          >
            bees
          </a>
        </p>
        <p>
          <b>
            Reaction:
          </b>
           hives
        </p>
        <p>
          <b>
            Notes:
          </b>
           Carry epipen.
        </p>
        <p />
      </div>
    </div>
  </li>
</DocumentFragment>
`;
